src/test/denialofservice_tests.cpp

```diff
@@ -223,47 +223,84 @@ BOOST_AUTO_TEST_CASE(peer_discouragement)
 {
     const CChainParams& chainparams = Params();
     auto banman = MakeUnique<BanMan>(GetDataDir() / "banlist.dat", nullptr, DEFAULT_MISBEHAVING_BANTIME);
-    auto connman = MakeUnique<CConnman>(0x1337, 0x1337);
+    auto connman = MakeUnique<CConnmanTest>(0x1337, 0x1337);
```

Comment:
nit: Would be nice to split the "rafactor" part from the "new-tests" part in this commit

Reply:
hahaha what

Reply:
Done and copied the tests to https://github.com/bitcoin/bitcoin/pull/20966 (since this PR is closed without merge).